classes/history_table.php

```diff
@@ -32,8 +32,10 @@ class history_table extends table_sql {
      /**
       * Create table
       * @param string $uniqueid
+      * @param int $startdate
+      * @param int $enddate
       */
-    public function __construct(string $uniqueid) {
+    public function __construct(string $uniqueid, int $startdate = 0, int $enddate = 0) {
```

Comment:
I would probably make these not defaults.

Otherwise as-is, if you didn't supply them the sql would search for `startdate = 0` and `enddate = 0` which would probably return no results.

I think in this case since the sql is over the logstore table the function should require date ranges

Reply:
Refactored in last commit